system/jlib/jptree.cpp

```diff
@@ -4016,7 +4016,7 @@ class CommonReaderBase : public CInterface
             c++;
         }
     }
-    void error(const char *msg=NULL, bool giveContext=true, PTreeReadExcptCode code=PTreeRead_syntax)
+    [[noreturn]] void error(const char *msg=NULL, bool giveContext=true, PTreeReadExcptCode code=PTreeRead_syntax)
```

Comment:
The c++11 way of adding __attribute__((x)).  Tested with VS2015, VS2017 new and older gccs.